packages/graphql/src/components/DeferredInterfaceValidation.ts

```diff
@@ -0,0 +1,249 @@
+import { Children, isRefkey, Refkey } from "@alloy-js/core";
+import { GraphQLOutputSymbol } from "../symbols/graphql-output-symbol.js";
+import { ref } from "../symbols/reference.js";
+
+interface FieldInfo {
+  name: string;
+  type: string;
+  args: Array<{ name: string; type: string }>;
+}
+
+interface ResolvedInterface {
+  name: string;
+  symbol: GraphQLOutputSymbol;
+}
+
+interface PendingValidation {
+  typeName: string;
+  typeSymbol: GraphQLOutputSymbol;
+  interfaces: ResolvedInterface[];
+}
+
+// Global registry for types that need validation
+const pendingValidations: PendingValidation[] = [];
+const validationErrors: Error[] = [];
+
+/**
+ * Extract field information from a symbol's member space
+ * Note: symbolNames map values are actually OutputSymbol objects, not refkeys
+ */
+function extractFieldInfo(symbol: GraphQLOutputSymbol): FieldInfo[] {
+  const fields: FieldInfo[] = [];
+
+  // Iterate through member names and their symbols
+  // The map values are OutputSymbol objects, not refkeys
+  for (const [fieldName, fieldSymbol] of symbol.members.symbolNames.entries()) {
+    try {
+      // fieldSymbol is already an OutputSymbol, not a refkey
+      if (fieldSymbol && (fieldSymbol as any).metadata) {
+        const metadata = (fieldSymbol as any).metadata;
```

Comment:
Hm I don't quite understand why the cast to `any` or the null check is needed. This smells like AI slop.